kpi/tests/api/v2/test_api_assets.py

```diff
@@ -1747,6 +1749,66 @@ def test_asset_redeployment(self):
         self.assertEqual(self.asset.deployment.version_id,
                          version_id)
 
+    def test_asset_deployment_dates(self):
+        p = dateutil.parser.parse
+
+        assert self.asset.date_deployed is None
+        asset_response = self.client.get(self.asset_url, format='json')
+        assert asset_response.data['date_deployed'] is None
+
+        before = timezone.now()
+        self.test_asset_deployment()
+        after = timezone.now()
+
+        asset_response = self.client.get(self.asset_url, format='json')
+        date_first_deployed = asset_response.data['date_deployed']
+        assert before <= p(date_first_deployed) <= after
+        deployed_version = asset_response.data['deployed_versions']['results'][
+            0
+        ]
+        assert asset_response.data['version_id'] == deployed_version['uid']
+        # why does this fail?
+        '''
+        assert(
+            asset_response.data['deployed_version_id']
+            == deployed_version['uid']
+        )
+        '''
```

Comment:
@noliveleger, `asset_response.data['deployed_version_id']` is still `None` here :flushed:
![image](https://github.com/kobotoolbox/kpi/assets/2085013/94f86966-7d02-4aff-8bbb-b09aa3966dfe)

Reply:
OK, it's because `MockDeploymentBackend.connect()` does not set `version` in the `_deployment_data`:
https://github.com/kobotoolbox/kpi/blob/f7e99fde4e49dc3ec6afcdd3954229ea83b3b9ca/kpi/deployment_backends/mock_backend.py#L146-L156

Compare to `KobocatDeploymentBackend`:
https://github.com/kobotoolbox/kpi/blob/f7e99fde4e49dc3ec6afcdd3954229ea83b3b9ca/kpi/deployment_backends/kobocat_backend.py#L322-L328